consensus/scp/src/slot.rs

```diff
@@ -494,10 +496,11 @@ impl<V: Value, ValidationError: Display> Slot<V, ValidationError> {
             // weight256 is the node's weight, scaled to 0..<max uint256>
             // (weight256 = <max uint256> * <num> / <denom>)
             let (num, denom) = self.weight(node_id);
-            let mut tmp = bigint::U512::from(bigint::U256::max_value());
-            tmp = tmp.saturating_mul(bigint::U512::from(num));
-            tmp = tmp.overflowing_div(bigint::U512::from(denom)).0;
-            let weight256 = bigint::U256::from(tmp);
+            let mut tmp = U512::from(U256::max_value());
+            tmp = tmp.saturating_mul(U512::from(num));
+            tmp = tmp / U512::from(denom);
+            let weight256 = U256::try_from(tmp)
+                .expect("failure calculating weight (max_u256 * k -> 2^512) / n");
```

Comment:
Good changeout for the crate, but why are does it use such huge numbers here?

Reply:
I don't know for certain but the function comment refers to "p.20 of the [Whitepaper](https://www.stellar.org/papers/stellar-consensus-protocol.pdf)," which describes how nodes are supposed to compute what I assume is intended as a deterministic random priority (using a SHA256 hash in the paper, Keccak in our case) for each of their neighbors.

Reply:
I'm sure it's correct, but I'm going to take a review of that.

Reply:
Neat, I'll take a review of that. I'm sure it's correct.